tests/integration/rules/countable.phpt

```diff
@@ -0,0 +1,70 @@
+--FILE--
+<?php
+require 'vendor/autoload.php';
+
+use Respect\Validation\Exceptions\AllOfException;
+use Respect\Validation\Exceptions\CountableException;
+use Respect\Validation\Validator as v;
+
+try {
+    v::not(v::countable())->check([]);
+} catch (CountableException $exception) {
+    echo $exception->getMessage().PHP_EOL;
+}
+
+try {
+    v::countable()->check(1.0);
+} catch (CountableException $exception) {
+    echo $exception->getMessage().PHP_EOL;
+}
+
+try {
+    v::countable()->check(PHP_INT_MAX);
+} catch (CountableException $exception) {
+    echo $exception->getMessage().PHP_EOL;
+}
+
+try {
+    v::countable()->check(true);
+} catch (CountableException $exception) {
+    echo $exception->getMessage().PHP_EOL;
+}
+
+try {
+    v::countable()->check(new \stdClass());
+} catch (CountableException $exception) {
+    echo $exception->getMessage().PHP_EOL;
+}
+
+try {
+    v::countable()->assert('Not countable!');
+} catch (AllOfException $exception) {
+    echo $exception->getFullMessage().PHP_EOL;
```

Comment:
The goal of the integration tests is to just test how the exceptions behave. That said we only need to test `check()`, `assert()`, `not(check())`, and `not(assert()`. Could you remove some extra tests and put them to the unit test instead -- if they are not there already?